EIPS/EIP-1776.md

```diff
@@ -0,0 +1,187 @@
+---
+eip: 1776
+title: ERC-721 Royalty Standard
+author: Zach Burks (@vexycats)
+discussions-to: https://github.com/ethereum/EIPs/issues/2907
+status: Draft
+type: Standards Track
+category: ERC
+created: 2020-09-15
+requires: 721
+---
+
+## Simple Summary
+
+A standardized way to handle royalty payments for ERC-721 tokens, including publically viewable information and notification of payment event.
+
+## Abstract
+
+Many of the largest ERC-721 marketplaces have implemented a form of royalties that is not compatible across the board and therefore worthless if the NFT is sold on another marketplace, defeating the purpose of any royalty system. This standard is a proposed way to minimally implement royalties that can be accepted across any type of NFT marketplace smart contracts. This standard allows for a royalty standard to be accepted on all marketplaces - leaving the funds transfer up to the marketplace itself, and only providing a means to fetch the royalty amounts and a event to be fired off when transfer has happened.  
+
+## Motivation
+
+This extension provides even more flexibility to the [ERC-721 specification](./eip-721.md). It is possible to set a royalty amount that can be paid to the creator on any marketplace that implements this ERC. If a marketplace chooses not to implement this ERC then of course no funds are paid for secondary sales. But seeing as how most NFT marketplace have developed some sort of royalty system themselves - and all of them are singular and only work on their own contracts - there needs to be an accepted standard way of providing royalties - if the creator so chooses to set royalties on their NFTs.
+Many decentralized marketplaces use different means to facilitate the trade of NFTs - therefore this ERC does not propose a method of transferring funds - but it simply provides a means to return the royalty amounts, and a function to be called once the transfer has been completed. Allowing each marketplace to easily integrate the standard in whatever flow they currently use for their marketplace.
+
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL
+NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and
+"OPTIONAL" in this document are to be interpreted as described in
+RFC 2119.
+
+**ERC-721 compliant contracts MAY implement this ERC for royalties to provide a standard method of accepting royalty payments and receiving royalty information**
+
+The `_RoyaltyAmount` **MUST** be calculated as a percentage - such as "5" - for 5%. This is **REQUIRED** to maintain uniformity across the standard. 
+
+Marketplaces that implement the `Royalty` standard **MAY** implement any method of calculation or transfer of funds depending on their marketplace style and smart contract code. 
+
+```solidity
+pragma solidity ^0.6.0;
+import "./ERC165.sol";
+
+/**
+ * @dev Implementation of royalties for 721s
+ *
+ */
+abstract contract Royalties is ERC165 {
```

Comment:
Yes thank you. As mentioned above it will be converted to a contract as this was my implementation and interface probably shouldn't be included for the standard

Reply:
> it will be converted to a contract as this was my implementation and interface probably shouldn't be included for the standard

Do you mean the other way around?  Interface standards should never have implementation details in the specification, though sometimes a reference implementation may be provided in the `## Implementation` section.